py_bentoctl/config.py

```diff
@@ -61,62 +79,95 @@ def _env_get_bool(var: str, default: bool = False) -> bool:
+        BENTO_OPTIONAL_FEATURES.append(self)
+        BENTO_OPTIONAL_FEATURES_BY_PROFILE[profile] = self
 
-def _filter_base_services(services: Iterable[str]) -> Tuple[str, ...]:
-    return tuple(service for service in services if service not in BASE_SERVICES)
 
+BENTOV2_USE_EXTERNAL_IDP: bool = _env_get_bool("BENTOV2_USE_EXTERNAL_IDP", default=False)
+BENTOV2_USE_BENTO_PUBLIC: bool = _env_get_bool("BENTOV2_USE_BENTO_PUBLIC", default=True)
+
+BENTO_FEATURE_AUTH = BentoOptionalFeature(enabled=not BENTOV2_USE_EXTERNAL_IDP, profile="auth")
+BENTO_FEATURE_BEACON = BentoOptionalFeature(
+    enabled=_env_get_bool("BENTO_BEACON_ENABLED", default=False), profile="beacon")
+BENTO_FEATURE_CBIOPORTAL = BentoOptionalFeature(
+    enabled=_env_get_bool("BENTO_CBIOPORTAL_ENABLED", default=False), profile="cbioportal")
+BENTO_FEATURE_GOHAN = BentoOptionalFeature(
+    enabled=_env_get_bool("BENTO_GOHAN_ENABLED", default=False), profile="gohan")
+BENTO_FEATURE_PUBLIC = BentoOptionalFeature(enabled=BENTOV2_USE_BENTO_PUBLIC, profile="public")
 
-def _get_optional_compose_services(feature_flag: bool, compose_file: str) -> Tuple[str, ...]:
-    if not feature_flag:
-        return ()
+if not DEV_MODE and BENTO_FEATURE_CBIOPORTAL.enabled:
+    import sys
+    print("cBioPortal for production deployments is not finished.", file=sys.stderr)
+    exit(1)
 
-    with open(compose_file) as cfh:
-        base_data = yaml.load(cfh, yaml.Loader)
 
-    return _filter_base_services(base_data["services"].keys())
+BENTO_GIT_CLONE_HTTPS: bool = os.getenv("BENTO_GIT_CLONE_HTTPS", "0").lower().strip() in ("1", "true")
 
+COMPOSE: Tuple[str, ...] = ("docker", "compose")
 
-# Load dev docker-compose services list if in DEV_MODE
-DOCKER_COMPOSE_DEV_SERVICES: Tuple[str, ...] = _get_optional_compose_services(DEV_MODE, DOCKER_COMPOSE_FILE_DEV)
 
-# Optional feature compose services start -----------------------------------------------
-DOCKER_COMPOSE_AUTH_SERVICES: Tuple[str, ...] = _get_optional_compose_services(
-    not BENTOV2_USE_EXTERNAL_IDP, DOCKER_COMPOSE_FILE_FEATURE_AUTH)
+def _get_enabled_services(compose_file: str, filter_out: Tuple[str, ...] = ()) -> Generator[str, None, None]:
+    # Loop through compose file and find enabled services - either no profiles specified,
+    # or the profile of an enabled feature.
+
+    # Load base docker-compose services list
+    with open(compose_file) as dcf:
+        compose_data = yaml.load(dcf, yaml.Loader)
+
+    for k, v in compose_data["services"].items():
+        if k in filter_out:
+            continue
+
+        profiles: Optional[List[str]] = v.get("profiles")
+        if profiles is None:
+            # If no profiles, always enabled
+            yield k
+            continue
+        for p in profiles:
+            if (f := BENTO_OPTIONAL_FEATURES_BY_PROFILE.get(p)) is not None and f.enabled:
+                yield k
+                break
+            if p == BENTO_PROFILE_DEV:
```

Comment:
Dont see any "dev" profiles in the compose yamls, what is the expected behaviour here?

Reply:
i think we maintain the status quo of separating prod/dev by docker-compose files, and only use profiles to toggle services

Reply:
removed